internal/ingress/annotations/conditions/types.go

```diff
@@ -0,0 +1,255 @@
+	// may be cached.
+	Values []*string
+}
+
+func (c *HttpRequestMethodConditionConfig) validate() error {
+	if len(c.Values) == 0 {
+		return errors.New("Values cannot be empty")
+	}
+	return nil
+}
+
+// Information about a path pattern condition.
+type PathPatternConditionConfig struct {
+	// One or more path patterns to compare against the request URL. The maximum
+	// size of each string is 128 characters. The comparison is case sensitive.
+	// The following wildcard characters are supported: * (matches 0 or more characters)
+	// and ? (matches exactly 1 character).
+	//
+	// If you specify multiple strings, the condition is satisfied if one of them
+	// matches the request URL. The path pattern is compared only to the path of
+	// the URL, not to its query string. To compare against the query string, use
+	// QueryStringConditionConfig.
+	Values []*string
+}
+
+func (c *PathPatternConditionConfig) validate() error {
+	if len(c.Values) == 0 {
+		return errors.New("Values cannot be empty")
+	}
+	return nil
+}
+
+// Information about a key/value pair.
+type QueryStringKeyValuePair struct {
+	// The key. You can omit the key.
+	Key *string
+
+	// The value.
+	Value *string
+}
+
+// Information about a query string condition.
+//
+// The query string component of a URI starts after the first '?' character
+// and is terminated by either a '#' character or the end of the URI. A typical
+// query string contains key/value pairs separated by '&' characters. The allowed
+// characters are specified by RFC 3986. Any character can be percentage encoded.
+type QueryStringConditionConfig struct {
+	// One or more key/value pairs or values to find in the query string. The maximum
+	// size of each string is 128 characters. The comparison is case insensitive.
+	// The following wildcard characters are supported: * (matches 0 or more characters)
+	// and ? (matches exactly 1 character). To search for a literal '*' or '?' character
+	// in a query string, you must escape these characters in Values using a '\'
+	// character.
+	//
+	// If you specify multiple key/value pairs or values, the condition is satisfied
+	// if one of them is found in the query string.
+	Values []*QueryStringKeyValuePair
+}
+
+func (c *QueryStringConditionConfig) validate() error {
+	if len(c.Values) == 0 {
+		return errors.New("Values cannot be empty")
+	}
+	return nil
+}
+
+// Information about a source IP condition.
+//
+// You can use this condition to route based on the IP address of the source
+// that connects to the load balancer. If a client is behind a proxy, this is
+// the IP address of the proxy not the IP address of the client.
+type SourceIpConditionConfig struct {
+	// One or more source IP addresses, in CIDR format. You can use both IPv4 and
+	// IPv6 addresses. Wildcards are not supported.
+	//
+	// If you specify multiple addresses, the condition is satisfied if the source
+	// IP address of the request matches one of the CIDR blocks. This condition
```

Comment:
Nit: `// IP address of the request matches any of the CIDR blocks.`

Reply:
The comments is copied directly from awsSDKGO :D